compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -441,7 +441,8 @@ pub fn compile_declarative_macro(
     let argument_gram = mbe::macro_parser::compute_locs(&sess.parse_sess, &argument_gram);
 
     let parser = Parser::new(&sess.parse_sess, body, true, rustc_parse::MACRO_ARGUMENTS);
-    let mut tt_parser = TtParser::new(def.ident);
+    let mut tt_parser =
+        TtParser::new(Ident::with_dummy_span(if macro_rules { kw::MacroRules } else { kw::Macro }));
```

Comment:
This is reasonable while we are still matching the macro definition against `$( $lhs:tt => $rhs:tt );+` reusing `macro_rules` machinery.

However, we really should stop pretending that `macro_rules! foo { ... }` or `macro foo (...) { ... }` are macro invocations.
This involves all the comments saying that, introducing a custom parser for the `$( $lhs:tt => $rhs:tt );+` syntax (and its `macro` 2.0 equivalent), and storing the output from such parser (the lowered macro representation) to crate metadata, so that macro definitions are not reparsed as raw token streams by every dependent crate.